### ISSUE TITLE
libelf deprecation on macOS

### ISSUE THREAD

**Issue by iMichka** [GNS3/dynamips#142]
Hello

I am a maintainer of the Homebrew project and wanted to give a small heads up: we would like to deprecate libelf on macOS.

The upstream libelf website is gone, the remaining source code can only be through archive.org ... (since at least 2019, which was 5 years ago ...). dynamips is one of the last projects that is using libelf in macOS.

Are there any plans for a replacement solution on your side?

**Comment by flaviojs** [comment]
If it's only a matter of changing the build system to use a compatible library or different libelf source, then I'm willing to do that. Just point me to what should be used.

If it requires changing C code, then only if someone adds [Coverity Scan](https://scan.coverity.com/).
I'm willing to fix what shows up there and then handle this too.

**Comment by iMichka** [comment]
I think it's harder than expected, see the ongoing discussion here: https://github.com/avrdudes/avrdude/issues/1653

**Comment by grossmj** [comment]
Indeed. Since Dynamips is the only project that needs libelf, maybe we could consider embed it in our repository and use it to build on macOS?

**Comment by flaviojs** [comment]
This feels like something that should be done in Homebrew...

The problem is libelf only being available through archive.org.
Dynamips already points users to elfutils-libelf-devel on Redhat based systems, so it's already a compatible replacement.
Why not have a formula for elfutils-libelf in Homebrew?

The whole of elfutils might be hard due to linux-only stuff, but the libelf in elfutils by itself should be doable.
Then Homebrew can deprecate libelf in favor of elfutils-libelf.
